### PR TITLE
Fixes #15676: Remove Map#keySet because of memory not freed

### PR DESCRIPTION
**Pull request by fanf**
https://issues.rudder.io/issues/15676

### CODE REVIEW

**Comment by fanf** [comment]
We will wait for `5.0.14` and real tests for that one.

File: webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/eventlog/HistorizationService.scala [hunk 1]
```diff
@@ -188,9 +188,9 @@ class HistorizationServiceImpl(
          }
       }.toSeq.map { case (t,fat,d) => (d, fat.toActiveTechnique, t) }
 
-      val stringDirectiveIds = directives.keySet.map( _.value)
+      val stringDirectiveIds = directives.keysIterator.toSet.map( _.value)
```
Comment (ncharles) [inline webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/eventlog/HistorizationService.scala]: wow, this fails 
[ERROR] /root/dev/rudder/webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/eventlog/HistorizationService.scala:191: error: missing parameter type for expanded function ((x$4: <error>) => x$4.value)
[ERROR]       val stringDirectiveIds = directives.keysIterator.toSet.map( _.value)
[ERROR]                                                                   ^

i have no idea why ...

Comment (ncharles) [reply webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/eventlog/HistorizationService.scala]: doing it in 2 steps fixes the issues
      val directiveSet = directives.keysIterator.toSet
      val stringDirectiveIds = directiveSet.map( _.value)

File: webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/quicksearch/QuickSearchBackendImpl.scala [hunk 2]
```diff
@@ -289,7 +289,7 @@ object QSLdapBackend {
 
     if(m.size != QSAttribute.all.size) {
       throw new IllegalArgumentException("Be carefull, it seems that the list of attributes in QSAttribute was modified, but not the list of name mapping" +
-          s"Please check for '${(m.keySet.diff(QSAttribute.all)++QSAttribute.all.diff(m.keySet)).mkString("', '")}'"
+          s"Please check for '${(m.keysIterator.toSet.diff(QSAttribute.all)++QSAttribute.all.diff(m.keysIterator)).mkString("', '")}'"
```
Comment (ncharles) [inline webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/quicksearch/QuickSearchBackendImpl.scala]: missing a toSet here

**Pull request closed without merging**